src/compiler/gcc.rs

```diff
@@ -303,6 +309,55 @@ where
         };
     }
 
+    let xclang_it = ExpandIncludeFile::new(cwd, &xclangs);
+    for arg in ArgsIter::new(xclang_it, (&ARGS[..], &clang::ARGS[..])) {
+        let arg = try_or_cannot_cache!(arg, "argument parse");
+        let args = match arg.get_data() {
+            Some(SplitDwarf) |
+            Some(ProfileGenerate) |
+            Some(TestCoverage) |
+            Some(Coverage) |
+            Some(DoCompilation) |
+            Some(Language(_)) |
+            Some(Output(_)) |
+            Some(TooHardFlag) |
+            Some(XClang(_)) |
+            Some(TooHard(_)) => {
+                cannot_cache!(arg.flag_str().unwrap_or(
+                    "Can't handle complex arguments through clang",
+                ))
+            }
+            None => {
+                match arg {
+                    Argument::Raw(_) => cannot_cache!("Can't handle Raw arguments with -Xclang"),
+                    Argument::UnknownFlag(_) => cannot_cache!("Can't handle UnknownFlag arguments with -Xclang"),
+                    _ => unreachable!(),
+                }
+            }
+            Some(PassThrough(_)) |
+            Some(PassThroughPath(_)) => Some(&mut common_args),
+            Some(PreprocessorArgumentFlag) |
+            Some(PreprocessorArgument(_)) |
+            Some(PreprocessorArgumentPath(_)) |
+            Some(DepTarget(_)) |
+            Some(NeedDepTarget) => Some(&mut preprocessor_args),
+        };
+
+        if let Some(args) = args {
+            // Normalize attributes such as "-I foo", "-D FOO=bar", as
+            // "-Ifoo", "-DFOO=bar", etc. and "-includefoo", "idirafterbar" as
+            // "-include foo", "-idirafter bar", etc.
+            let norm = match arg.flag_str() {
+                Some(s) if s.len() == 2 => NormalizedDisposition::Concatenated,
+                _ => NormalizedDisposition::Separated,
+            };
+            for arg in arg.normalize(norm).iter_os_strings() {
+                args.push("-Xclang".into());
+                args.push(arg)
+            }
+        }
+    }
+
```

Comment:
This code is identical to the code in the MSVC implementation, right?
https://github.com/mozilla/sccache/blob/e722f83b3eb3e3f6377524fd6eb20e72d57ee1d6/src/compiler/msvc.rs#L325-L373

Should we factor this out into a separate function that we could share?